include/vinecopulib/bicop/class.hpp

```diff
@@ -119,24 +115,40 @@ class Bicop
 
   Eigen::MatrixXd get_parameters_upper_bounds() const;
 
+  Bicop as_continuous() const;
+
 private:
-  Eigen::Matrix<double, Eigen::Dynamic, 2> cut_and_rotate(
-    const Eigen::Matrix<double, Eigen::Dynamic, 2>& u) const;
+  Eigen::MatrixXd format_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd clip_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd rotate_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd prep_for_abstract(const Eigen::MatrixXd& u) const;
 
   void check_rotation(int rotation) const;
 
+  void check_data(const Eigen::MatrixXd& u) const;
+
+  void check_data_dim(const Eigen::MatrixXd& u) const;
+
+  void flip_var_types();
+
   void check_weights_size(const Eigen::VectorXd& weights,
                           const Eigen::MatrixXd& data) const;
 
   void check_fitted() const;
 
+  int get_n_discrete() const;
+
   double compute_mbic_penalty(const size_t nobs, const double psi0) const;
 
   BicopPtr get_bicop() const;
 
   BicopPtr bicop_;
   int rotation_;
-  size_t nobs_;
+  size_t nobs_{0};
+  mutable std::vector<std::string> var_types_{ "c", "c" };
```

Comment:
Why are you adding this one a second time? Isn't it in abstract already?

Unrelated, but now that I think about it, why is `loglik` in abstract and `nobs` here?